### PR TITLE
Better trash detection

### LINKED ISSUE #229: Possible to drag & drop within trash
Regression?  It should not be possible to drag and drop files around inside the trash.

### PR DESCRIPTION
**Pull request by jeremypw**
Fixes #229 

* New Vala PF.FileUtils.function location_is_in_trash ()
* Use new function throughout.

Still detects file is in trash even if the scheme is not trash:// but the actual location of the system or user trash folder.

### CODE REVIEW

File: libcore/FileUtils.vala [hunk 1]
```diff
@@ -552,6 +552,15 @@ namespace PF.FileUtils {
                 return false;
         }
     }
+
+    public bool location_is_in_trash (GLib.File location) {
```
Comment (donadigo) [inline libcore/FileUtils.vala]: This should be static.

File: libcore/FileUtils.vala [hunk 2]
```diff
@@ -552,6 +552,15 @@ namespace PF.FileUtils {
...
+        var uri = location.get_uri ();
+
+        return (uri.has_prefix ("trash") ||
```
Comment (donadigo) [inline libcore/FileUtils.vala]: Shouldn't that test if the file has an URI scheme, like the C code did?

Comment (jeremypw) [reply libcore/FileUtils.vala]: I guess so - this is just a bit quicker.

**Comment by jeremypw** [comment]
I found that the uri passed to DnD can lose the trash:// scheme e.g. by opening folders in trash. If trashed files are not properly detected then the permitted DnD operations will not be correct.  I guess this could be regarded as a "work around" and we should try and make sure that only trash:// scheme uris are generated.

Is there a downside to keeping the extra checks, at least temporarily?

**Pull request merged**